server/testnet.go

```diff
@@ -0,0 +1,240 @@
+
+				nodeDirName := cmn.Fmt("%s%d", nodeDirPrefix, i)
+				nodeDir := filepath.Join(outputDir, nodeDirName, "gaiad")
+				clientDir := filepath.Join(outputDir, nodeDirName, "gaiacli")
+				gentxsDir := filepath.Join(outputDir, "gentxs")
+				gaiaConfig := gaiacfg.DefaultConfig()
+				gaiaConfig.Name = nodeDirName
+				gaiaConfig.CliRoot = clientDir
+				gaiaConfig.Init.ChainID = "something-something"
+				gaiaConfig.Init.GenTxs = true
+				gaiaConfig.Init.Overwrite = true
+				gaiaConfig.Init.GenTxsDir = gentxsDir
+				config.SetRoot(nodeDir)
+
+				// Run `init` and generate genesis.json
+				_, _, _, err := initWithConfig(config,gaiaConfig, ctx,cdc,appInit)
+				if err != nil {
+					return err
+				}
+
+				pvFile := filepath.Join(nodeDir, config.BaseConfig.PrivValidator)
+				pv := pvm.LoadFilePV(pvFile)
+				genVals[i] = types.GenesisValidator{
+					PubKey: pv.GetPubKey(),
+					Power:  1,
+					Name:   nodeDirName,
+				}
+			}
+
+			for i := 0; i < nNonValidators; i++ {
+				nodeDirName := cmn.Fmt("%s%d", nodeDirPrefix, i+nValidators)
+				nodeDir := filepath.Join(outputDir, nodeDirName, "gaiad")
+				config.SetRoot(nodeDir)
+
+				err := os.MkdirAll(filepath.Join(nodeDir, "config"), nodeDirPerm)
+				if err != nil {
+					_ = os.RemoveAll(outputDir)
+					return err
+				}
+
+				config.Moniker = nodeDirName
+				//TODO: Fix it!v
+				initWithConfig(config,nil,ctx,cdc,appInit)
+			}
+
+			if populatePersistentPeers {
+				err := populatePersistentPeersInConfigAndWriteIt(config)
+				if err != nil {
+					_ = os.RemoveAll(outputDir)
+					return err
+				}
+			}
+
+			fmt.Printf("Successfully initialized %v node directories\n", nValidators+nNonValidators)
+			return nil
+			},
+	}
+	cmd.Flags().IntVar(&nValidators, "v", 4,
+		"Number of validators to initialize the testnet with")
+	cmd.Flags().IntVar(&nNonValidators, "n", 0,
+		"Number of non-validators to initialize the testnet with")
+	cmd.Flags().StringVar(&outputDir, "o", "./mytestnet",
+		"Directory to store initialization data for the testnet")
+	cmd.Flags().StringVar(&nodeDirPrefix, "node-dir-prefix", "node",
+		"Prefix the directory name for each node with (node results in node0, node1, ...)")
+
+	cmd.Flags().BoolVar(&populatePersistentPeers, "populate-persistent-peers", false,
+		"Update config of each node with the list of persistent peers build using either hostname-prefix or starting-ip-address")
+	cmd.Flags().StringVar(&hostnamePrefix, "hostname-prefix", "node",
+		"Hostname prefix (node results in persistent peers list ID0@node0:46656, ID1@node1:46656, ...)")
+	cmd.Flags().StringVar(&startingIPAddress, "starting-ip-address", "",
+		"Starting IP address (192.168.0.1 results in persistent peers list ID0@192.168.0.1:46656, ID1@192.168.0.2:46656, ...)")
+	cmd.Flags().IntVar(&p2pPort, "p2p-port", 46656,
+		"P2P Port")
+	cmd.AddCommand(GenTxCmd(ctx, cdc, appInit))
+	return cmd
+}
+
+func hostnameOrIP(i int) string {
+	if startingIPAddress != "" {
```

Comment:
rather than indenting the entire function for a single if, just flip it and return at the beginning:

```
hostname := fmt.Sprintf("%s%d", hostnamePrefix, i)
if startingIPAddress == "" {
    return hostname
}
... the other code
...
return hostname
```